pkg/codegen/dotnet/gen.go

```diff
@@ -2158,11 +2158,20 @@ func genPackageMetadata(pkg *schema.Package,
 	if err != nil {
 		return err
 	}
-	plugin, err := (&plugin.PulumiPluginJSON{
+
+	pulumiPlugin := &plugin.PulumiPluginJSON{
 		Resource: true,
 		Name:     pkg.Name,
 		Server:   pkg.PluginDownloadURL,
-	}).JSON()
+	}
+
+	lang, ok := pkg.Language["csharp"].(CSharpPackageInfo)
+	if pkg.Version != nil && ok && lang.RespectSchemaVersion {
+		files.add("version.txt", []byte(pkg.Version.String()))
+		pulumiPlugin.Version = pkg.Version.String()
```

Comment:
Should we do the same for Go and Python's `pulumi-plugin.json` file if we're doing it for .NET?